src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -59,7 +59,9 @@
         packed[ArgumentIndicies.SELF.ordinal()] = self;
         packed[ArgumentIndicies.BLOCK.ordinal()] = block;
 
-        ArrayUtils.arraycopy(arguments, 0, packed, RUNTIME_ARGUMENT_COUNT, arguments.length);
+        if (arguments != null) {
+            ArrayUtils.arraycopy(arguments, 0, packed, RUNTIME_ARGUMENT_COUNT, arguments.length);
+        }
```

Comment:
Why not use an singleton empty `Object[]` like the previous commits?
This might cost performance if the compiler doesn't know if `arguments` is null or not.

Reply:
Also `arguments` is not optional. We can pass 0 arguments of course, but so `null` doesn't seem a valid value, just a hack to avoid the allocation.
We could make `pack` take `Object...` for convenience but that seems bad since it has many arguments and could easily introduce subtle bugs.

Reply:
I was looking to simplify the caller (e.g., we already allow `null` when constructing nodes) and I wanted to avoid this `arraycopy` for situations that should just no-op anyway.